zebra-chain/src/primitives/address.rs

```diff
@@ -26,14 +25,8 @@ pub enum Address {
         /// Address' network
         network: Network,
 
-        /// Transparent address
-        transparent_address: transparent::Address,
-
-        /// Sapling address
-        sapling_address: sapling::PaymentAddress,
-
-        /// Orchard address
-        orchard_address: orchard::Address,
+        /// Unified address
+        address: zcash_address::unified::Address,
```

Comment:
This seems ok for now, but I think PR #6171 will want to expand it back out again. (The conversion from an address list to fields is already in that PR.)

Reply:
Actually, it might be easier doing it this way.

Reply:
I think we'll need to convert the addresses to match zcashd's behaviour (see https://github.com/zcash/zcash/blob/master/src/key_io.cpp#L448 and https://github.com/zcash/zcash/blob/master/src/rust/src/address_ffi.rs#L148-L152).

Reply:
I've been trying to find suitable UAs that would fail the check, but I couldn't find any. I have only valid UAs. :/

Reply:
I'm not sure if it's necessary to do it for this PR but I think we could use [`try_from_items`](https://docs.rs/zcash_address/latest/zcash_address/unified/trait.Encoding.html#method.try_from_items) for any random "address" values/items with the correct length and encoding.

Reply:
This PR is on the critical path for miner testing, so I'd like to get it merged soon.
Can we open a separate ticket for the unified address change?

Here's a previous conversation where Pili suggests we split out unified address support entirely:
https://github.com/ZcashFoundation/zebra/issues/6083#issuecomment-1432260991

Reply:
A suitable UA support is implemented in [d47abd1](https://github.com/ZcashFoundation/zebra/pull/6185/commits/d47abd133fcda832b16999ca929951934e4192e9).